subsys/net/lib/lwm2m/lwm2m_obj_firmware_pull.c

```diff
@@ -478,7 +479,20 @@ static void firmware_transfer(struct k_work *work)
 	tmp = server_addr[off + len];
 	server_addr[off + len] = '\0';
 
-	ret = net_app_init_udp_client(&firmware_ctx.net_app_ctx, NULL, NULL,
+	/* setup the local firmware download client port */
+	memset(&client_addr, 0, sizeof(client_addr));
+#if defined(CONFIG_NET_IPV6)
```

Comment:
Shouldn't we set the local family depending on what is the IP address type of the server?

Reply:
Same as above.  The netapp layer doesn't actually use the IP type if the IP address is unassigned.